bpi.syndicate.inc

```diff
@@ -543,3 +688,217 @@ function bpi_syndicate_action($bpi_id) {
+      '#type' => 'fieldset',
+      '#title' => t('Filters'),
+      '#collapsible' => TRUE,
+      '#collapsed' => !_bpi_is_filter_applied(),
+      '#attributes' => array(
+        'class' => array(
+          'bpi-filters',
+        ),
+      ),
+    );
+
+    $facets_form = array(
+      '#type' => 'container',
+      '#tree' => TRUE,
+      '#attributes' => array(
+        'class' => array(
+          'bpi-facets',
+        ),
+      ),
+    );
+
+    $use_chosen = TRUE;
+
+    if ($use_chosen) {
+      $form['#attached']['css'][] = drupal_get_path('module', 'bpi') . '/js/chosen/chosen.min.css';
+      $form['#attached']['js'][] = drupal_get_path('module', 'bpi') . '/js/chosen/chosen.jquery.min.js';
+    }
+
+    foreach ($facets as $name => $terms) {
+      $id = $name;
+      $type = $use_chosen ? 'select' : 'checkboxes';
+      $multiple = TRUE;
+      $options = array();
+      foreach ($terms as $term => $item) {
+        $options[$term] = $item['title'];
+      }
+      $default_value = _bpi_get_query_parameter(array('filter', $name), array());
+
+      switch ($name) {
+        case 'agency_internal':
+          $options = array(
+            '' => t('All'),
+          );
+          if (isset($terms[0])) {
+            $options['false'] = t('No');
+          }
+          if (isset($terms[1])) {
+            $options['true'] = t('Yes');
+          }
+          $default_value = _bpi_get_query_parameter(array('filter', $name));
+          $multiple = FALSE;
+          $type = 'select';
+          break;
+      }
+
+      $facets_form[$id] = array(
+        '#type' => $type,
+        '#multiple' => $multiple,
+        '#chosen' => $multiple && $use_chosen,
+        '#size' => $multiple ? min(20, count($options)) : NULL,
+        '#title' => bpi_label_mapper($name) ? bpi_label_mapper($name) : $name,
+        '#options' => $options,
+        '#default_value' => $default_value,
+        '#attributes' => array(
+          'data-placeholder' => $multiple ? t('Pick some values') : t('Pick a value'),
+          'data-no-matches' => t('Nothing matches'),
+        ),
+        '#prefix' => '<div class="bpi-facet bpi-facet-' . drupal_html_class($name) . '">',
+        '#suffix' => '</div>',
+      );
+    }
+
+    // Add logical operator.
+    $name = 'logicalOperator';
+    $id = $name;
+    $facets_form[$id] = array(
+      '#type' => 'select',
+      '#title' => bpi_label_mapper($name) ? bpi_label_mapper($name) : $name,
+      '#options' => array(
+        'and' => t('All must apply'),
+        'or' => t('Any must apply'),
+      ),
+      '#default_value' => _bpi_get_query_parameter(array('filter', $name), 'and'),
+      '#weight' => -10,
+    );
+
+    $filter_form['bpi_facets'] = $facets_form;
+
+    $filter_form['bpi_actions'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => array(
+          drupal_html_class('container-inline'),
+        ),
+      ),
+    );
+
+    if (_bpi_get_current_filter()) {
```

Comment:
_bpi_get_current_filter() returns as default an empty array, so should this be !empty() ?